Backport: endace: Fix source-dag timestamps

sfd opened the pull request:
Bug: [#6618](https://redmine.openinfosecfoundation.org/issues/6620)

Fix Endace ERF to SCTime_t timestamp conversion

Fix typo preventing compilation with --enable-dag

(cherry picked from commit https://github.com/OISF/suricata/commit/879db3dbc3e93912c784375c85d88404a9371f31)

Fix compiler const warning with --enable-dag

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [6620](https://redmine.openinfosecfoundation.org/issues/6620)

Updates PR https://github.com/OISF/suricata/pull/10092

sfd commented:
Not sure what is up with the Rust check.The Debian 12 build might need to be re-run.

sfd commented:
@victorjulien not sure what is wrong with the Rust test? The debian build probably just needs to be rerun.

catenacyber commented:
> not sure what is wrong with the Rust test? The debian build probably just needs to be rerun.

You should rebase to latest master, and it should go away

sfd commented:
> "Bug: 6620" is not a good commit title... And there is no cherry-pick line for this one... Do you know why ?

Updated the commit title.

It's not a cherry pick since the equivalent master PR is pending, see PR 10149. Could wait for that to be approved and cherry pick instead, but is equivalent? The compiler warning was noticed and fixed in the backport PR.

catenacyber commented:
> > "Bug: 6620" is not a good commit title... And there is no cherry-pick line for this one... Do you know why ?
> 
> Updated the commit title.
> 
> It's not a cherry pick since the equivalent master PR is pending, see PR 10149. Could wait for that to be approved and cherry pick instead, but is equivalent? The compiler warning was noticed and fixed in the backport PR.

I think we want to merge first in master, and then cherry pick, so this PR is waiting on 10149.
The rationale is that even if code is equivalent, we gets a better git history.

sfd commented:
Can do it that way if you prefer.
The alternative would be to back out the compiler warning commit and merge this PR, thereby bringing main-7.0.x up to date with master, then merge 10149 to master, and finally backport that as a separate PR?
This PR is more important since it fixes actual bugs in 7.x as opposed to just compiler warnings.

victorjulien commented:
> @sfd thanks for your work.
> 
> I am sorry I am not an expert on Endace... I just know a little about the GitHub workflow for Suricata, and why the CI was red for rust... I guess we will have to wait for someone with the endace knowledge to do the approvals and merge in the order they see fit...

@catenacyber @sfd is the Endace expert as he represents Endace here. We can trust that his Endace specific logic is correct, but can still review on style and process.

catenacyber commented:
So, now that https://github.com/OISF/suricata/pull/10149 was merged, could you make a new PR with the first commit of this one + a cherry-pick of the one from 10149 merged in master ? How does that sound to you ?

sfd commented:
Closed ; continued in PR #10216
